src/app/api/api_v1/endpoints/code_generation.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any
+from fastapi import APIRouter, HTTPException
+from fastapi.params import Depends
+from pydantic import BaseModel
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.api import deps
+from app.crud.crud_generation_template import generation_template
+from app.crud.crud_welding_configuration import welding_configuration
+from app.codegen.code_generator import CodeGenerator
+
+
+router = APIRouter()
+
+
+class RequestBodyGenerate(BaseModel):
+    generation_template_id: int
+    welding_configuration_id: int
+
+
+@router.post("/generate", response_model=str, deprecated=True)
```

Comment:
Why deprecated? Just in wise foresight that we will soon only need the `WeldingConfiguration` id? :D